### PR TITLE
AO3-4401 Batch readings rake task

### PR DESCRIPTION
**Pull request by zz9pzza**
Check that readings still happen, this is going to be very hard to tell if it is in effect or not.

I can look at the logs and that is about it.

If it passes the automated tests around readings then i wouldn't expect more testing than that.

### CODE REVIEW

**Comment by cesy** [comment]
https://github.com/otwcode/otwarchive/blob/master/features/other_a/reading.feature seems to cover the basics, but I can't see anything for updating the date and count when a work is read more than once - is that worth adding?

**Comment by zz9pzza** [comment]
do you mean checking " I can't see anything for updating the date and count when a work is read more than once - is that worth adding?"

or being confident that it does it ? I am pretty sure the code does the right thing as it sorts the list and runs though them in order.

**Comment by cesy** [comment]
I mean I can't see a test for that in our spec - I'm pretty confident that this code change won't affect it anyway, but it jumped out at me as a possible test coverage gap, since most other key functionality is in there. I was thinking about test coverage because you mentioned that we'll rely on the automated tests for this one, and that's something I could add fairly easily if we wanted it.

**Comment by zz9pzza** [comment]
I will happily take a pull for more tests :)

**Comment by cesy** [comment]
Cross-linking: #2266

File: app/models/reading.rb [hunk 1]
```diff
@@ -21,8 +21,12 @@ def self.mark_to_read_later(work, user)
 
   # called from rake
   def self.update_or_create_in_database
-    REDIS_GENERAL.smembers("Reading:new").reverse.each do |reading_json|
-      Reading.reading_object(reading_json)
+    REDIS_GENERAL.smembers("Reading:new").reverse.each_slice(ArchiveConfig.READING_BATCHSIZE || 1000) do |batch|
```
Comment (sarken) [inline app/models/reading.rb]: Is the plan to use a config setting or the fallback of 1000?

Comment (zz9pzza) [inline app/models/reading.rb]: The plan would be to use the fall back of 1000. I have started using that pattern for small bits of config.

File: app/models/reading.rb [hunk 2]
```diff
@@ -21,8 +21,12 @@ def self.mark_to_read_later(work, user)
...
+      Reading.transaction do
```
Comment (scottsds) [inline app/models/reading.rb]: Aren't you missing a variable here? 'Reading.transaction do |some_variable|

Comment (zz9pzza) [inline app/models/reading.rb]: No this is a database transaction

**Pull request merged**